Der hauptmann patch 1

DerHauptmann opened the pull request.

DerHauptmann committed:
Add files via upload

DerHauptmann committed:
Code was changed from cv2 and PIL to skimage

npeschke commented on Otsu_Thresholding.py:
```diff
@@ -0,0 +1,65 @@
+Global Otsu thresholding
+@author: mostly Luis
+"""
+
```
The file should start with the encoding (although this is not necessary anymore in python3), then comes the info on the file (author etc.) and finally the imports right infront of the code

npeschke commented on Otsu_Thresholding.py:
```diff
@@ -0,0 +1,65 @@
+@author: mostly Luis
+"""
+
+def import_image(path, name):
```
"path" commonly means the actual path to the file, something like 
`C:\users\bradypus\pictures\holiday.jpg`
What you mean in this context is the directory.

In line 14: What you mean by raw_image_directory is the actual path to the file.

npeschke commented on Otsu_Thresholding.py:
```diff
@@ -0,0 +1,65 @@
+import matplotlib.pyplot as plt
+from skimage.io import imread
```
If it is possible refrain from using the `from module import function` syntax. While it works like a charm, it is harder to comprehend.
`thresh = threshold_otsu(image)`
Take this for example, without knowing that you imported this function explicitly from `skimage.filters` I would expect that you wrote this function for yourself.

npeschke commented on Otsu_Thresholding.py:
```diff
@@ -0,0 +1,65 @@
+    ax[2].axis('off')
+
+    plt.show()
+    return
```
In case you don't want to return something, just omit the return keyword

npeschke commented on Otsu_Thresholding.py:
```diff
@@ -0,0 +1,65 @@
+    return image
+
+
+def figure(original_image, binary_image, threshold_value):
```
A more descriptive name might be appropriate instead of `figure`.

Also, this function does more than one thing:

1. It creates a figure with three subplots
2. It plots the original image
3. It plots a histogram with the threshold
4. It plots the binary image

Each of those functionalities should be handeled by their own respective functions like:

`def plot_image(axes, image, title)`
This can plot general images (original and binary) onto the specified axes object.
`def plot_otsu_hist(axes, image, otsu_threshold)`

npeschke commented on Otsu_Thresholding.py:
```diff
@@ -0,0 +1,65 @@
+def main():
+    # Directory and image were defined
+    name = "jw-1h1_c5.TIF"
+    path = "C:\\Users\\User\\PycharmProjects\\Otsu_thresholding\\"
```
Please add the images into the data directory of the repository but be aware of the 100 MB push limit of GitHub, meaning that a push can not contain more than 100 MB. You would exeed that if you add the files unzipped. So maybe keep them zipped and add some code to read them from the zipped file.

DerHauptmann committed:
set encoding in front and import rights after the script info

DerHauptmann committed:
corrected formal flaws:
raw image directory was changed to raw image path
from functions were taken out of import rights
useless return was deleted

DerHauptmann committed:
The code for the figure creation was modulated into individual functions.

DerHauptmann committed:
Creation of valuation via dice score

DerHauptmann committed:
implementation of valuation module

DerHauptmann committed:
cleaned version of implementation+ todo list

DerHauptmann committed:
image creation was changed to fig, ax function.
control is assembeling the right images only by name input(one variable).

DerHauptmann committed:
image creation was changed to fig, ax function.
control is assembeling the right images only by name input(one variable).

DerHauptmann committed:
zips are added

DerHauptmann committed:
zipped images are added

DerHauptmann committed:
changed module name. created new module for data management. linked control to loaded image

DerHauptmann committed:
added TODOs, polishing minor flaws

DerHauptmann committed:
All images were thresholded and error prone data(15min 5_4 43) was deleted

DerHauptmann committed:
All images were thresholded and error prone data(15min 5_4 43) was deleted

DerHauptmann committed:
all images are shown in one figure and total dice score is calculated

DerHauptmann committed:
cleaned version

DerHauptmann committed:
unzipping is packed im one function and figure is packed in function

DerHauptmann committed:
dice_calculation function and dice.main were unnecessary and deleted,
added docstring
adaptive name criteria was added
cleaning code

DerHauptmann committed:
image jw_30min have no controls and were deleted

DerHauptmann committed:
dice score subplot was visualized nice

DerHauptmann committed:
struggling with local otsu

DerHauptmann committed:
changed dice score subplot being f string generated and added score-dependent colours

DerHauptmann committed:
created a module to determine to determine a propper radius for local otsu
optimized image import for local otsu
added local otsu to main figure

DerHauptmann committed:
created a figure showing local otsu radius dependent score (ZIP->zip)
added docstrings
polished

DerHauptmann committed:
edge and filter determination (change of value) is created

DerHauptmann committed:
uniforming
changed output from useless plot to text

DerHauptmann committed:
modularized colouring of score increase

DerHauptmann committed:
modularized image path and name list acquisition.

DerHauptmann committed:
added one result figure that takes long to make

DerHauptmann committed:
Changed function names
+cleaning

DerHauptmann committed:
joining filtering, edging, local
deleting valuation

DerHauptmann committed:
added nuclei segmentation and counting

DerHauptmann committed:
visualized segmentation with label2rgb function

DerHauptmann committed:
cleaned
added doc strings

DerHauptmann committed:
divided filters in two functions
added otsu_report (notebook)
added median filter to Otsu Thresholding

DerHauptmann committed:
cleaning otsu_report

DerHauptmann committed:
Adjustance of names and titles
export of environment

DerHauptmann committed:
updated hist parameter ymax to top

DerHauptmann committed:
added right environment

DerHauptmann committed:
added anti-synergistic effect of improvements
and more text

DerHauptmann committed:
final

DerHauptmann committed:
final
